chainer/links/connection/lstm.py

```diff
@@ -197,6 +211,11 @@ def __call__(self, x):
             ~chainer.Variable: Outputs of updated LSTM units.
 
         """
+        if self.upward.has_uninitialized_params:
+            in_size = x.size // len(x.data)
+            self.upward._initialize_params(in_size)
```

Comment:
Is there any cases where the user might want to call it directly? Maybe if writing another subclass of LSTMBase? If so, it may be ok to make public.